infrastructure/src/main/java/org/corfudb/infrastructure/protocol/NettyServerRouter.java

```diff
@@ -96,14 +97,13 @@ public void channelActive(ChannelHandlerContext ctx) throws Exception {
     public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception {
         ByteBuf msgBuf = (ByteBuf) msg;
 
-        // Temporary -- Check Corfu msg marker: 0x1 indicates legacy while 0x2 indicates new
-        if (msgBuf.getByte(msgBuf.readerIndex()) == 0x1) {
-            ctx.fireChannelRead(msgBuf); // Forward legacy corfu msg to next handler
+        // Temporary -- If message is not a new Protobuf message, forward the message.
+        if (msgBuf.getByte(msgBuf.readerIndex()) != API.PROTO_CORFU_MSG_MARK) {
+            ctx.fireChannelRead(msgBuf);
             return;
         }
 
-        msgBuf.readByte(); // Temporary -- Consume 0x2 marker
-
+        msgBuf.readByte();
```

Comment:
What if this byte is not LEGACY_CORFU_MSG_MARK ?

I think you should explicitly check that it is LEGACY_CORFU_MSG_MARK and throw an IllegalStateException otherwise.

Reply:
Do you mean throw an exception (instead of passing it along the pipeline) if the message isn't marked with PROTO_CORFU_MSG_MARK?